packages/fiori/src/ViewSettingsDialog.ts

```diff
@@ -447,10 +453,12 @@ class ViewSettingsDialog extends UI5Element {
 			filters: this.filterItems.map(item => {
 				return {
 					text: item.text || "",
+					itemKey: item.itemKey || "",
 					selected: false,
 					filterOptions: item.values.map(optionValue => {
 						return {
 							text: optionValue.text || "",
+							itemKey: optionValue.itemKey || "",
```

Comment:
Change to: `item: optionValue`

Reply:
I believe I can even remove that line, as we can just return the slotted `filterItems` in the `get eventsParams()`. This way the app developers could execute their own logic for filtering the selected items based on their HTML Property they have provided (f.e `data-key`);